### ISSUE TITLE
[BUG] Investigate job errors

### ISSUE THREAD

**Issue by subdavis** [Kitware/dive#750]
These errors may spin off new issues

- [ ] Error in the logs at the bottom of https://viame.kitware.com/girder#job/60996e9383dfc8ded35dd2ca
- [ ] Transcoding for big buck bunny avi https://viame.kitware.com/girder#job/609beb6be30454c2d293cb0d
- [ ] 2 frames left in the queue on closing (below)

-----
# 2 frames left in the queue on closing

Transcoding failures of many videos: 

Some of this data looks strange, where there's a split in the video like this: https://viame.kitware.com/#/viewer/60e216f956bd9a6bdb485b9e

This appears to be related to an ffmpeg failure, but it may also indicate a problem with your video files.  Here's a quote from https://stackoverflow.com/questions/49686244/ffmpeg-too-many-packets-buffered-for-output-stream-01

> What is basically happening is there's a problem with how the source file video frames are structured, and ffmpeg is trying to keep track of the early frames in the file until it can figure out how it's all supposed to be organized. 

Here's an example of the failure log https://viame.kitware.com/girder#job/60e2ec2a474f6c55085bdc13

**Comment by BryonLewis** [comment]
TLDR:  Looks like the big buck bunny video is using `libaom-av1` for the AV1 codec which the built in container version of ffmpeg doesn't support.  

The first issues I think you mentioned were caused by corrupted video data.  I believe the big buck bunny issue is do to a missing codec in the ffmpeg version for the containers.

Below I just kept a running log of my thought process and exploration.
Running it in the container event with a simple default transcode I get:
```
root@d1bacb44bb42:/tmp# ffmpeg -i tmp0z1yp77_.mp4 output.mp4
ffmpeg version 3.4.8-0ubuntu0.2 Copyright (c) 2000-2020 the FFmpeg developers
  built with gcc 7 (Ubuntu 7.5.0-3ubuntu1~18.04)
  configuration: --prefix=/usr --extra-version=0ubuntu0.2 --toolchain=hardened --libdir=/usr/lib/x86_64-linux-gnu --incdir=/usr/include/x86_64-linux-gnu --enable-gpl --disable-stripping --enable-avresample --enable-avisynth --enable-gnutls --enable-ladspa --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libflite --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libgme --enable-libgsm --enable-libmp3lame --enable-libmysofa --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-librubberband --enable-librsvg --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libssh --enable-libtheora --enable-libtwolame --enable-libvorbis --enable-libvpx --enable-libwavpack --enable-libwebp --enable-libx265 --enable-libxml2 --enable-libxvid --enable-libzmq --enable-libzvbi --enable-omx --enable-openal --enable-opengl --enable-sdl2 --enable-libdc1394 --enable-libdrm --enable-libiec61883 --enable-chromaprint --enable-frei0r --enable-libopencv --enable-libx264 --enable-shared
  libavutil      55. 78.100 / 55. 78.100
  libavcodec     57.107.100 / 57.107.100
  libavformat    57. 83.100 / 57. 83.100
  libavdevice    57. 10.100 / 57. 10.100
  libavfilter     6.107.100 /  6.107.100
  libavresample   3.  7.  0 /  3.  7.  0
  libswscale      4.  8.100 /  4.  8.100
  libswresample   2.  9.100 /  2.  9.100
  libpostproc    54.  7.100 / 54.  7.100
[mov,mp4,m4a,3gp,3g2,mj2 @ 0x5570d14ce8c0] Format mov,mp4,m4a,3gp,3g2,mj2 detected only with low score of 1, misdetection possible!
[mov,mp4,m4a,3gp,3g2,mj2 @ 0x5570d14ce8c0] moov atom not found
tmp0z1yp77_.mp4: Invalid data found when processing input
```
From my basic reading the moov atom error is related to how the video was transcoded/created and there being and issue there.
My local version of ffmpeg will transcode the video properly but that its probably due to differences in the ffmpeg version.  It may have to do with the libaom support for 
for reference:
```
ffmpeg -i Big_Buck_Bunny_720_10s_10MB_AV1.mp4 out.mp4
ffmpeg version 4.2.4-1ubuntu0.1 Copyright (c) 2000-2020 the FFmpeg developers
  built with gcc 9 (Ubuntu 9.3.0-10ubuntu2)
  configuration: --prefix=/usr --extra-version=1ubuntu0.1 --toolchain=hardened --libdir=/usr/lib/x86_64-linux-gnu --incdir=/usr/include/x86_64-linux-gnu --arch=amd64 --enable-gpl --disable-stripping --enable-avresample --disable-filter=resample --enable-avisynth --enable-gnutls --enable-ladspa --enable-libaom --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libcodec2 --enable-libflite --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libgme --enable-libgsm --enable-libjack --enable-libmp3lame --enable-libmysofa --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-librsvg --enable-librubberband --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libssh --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwavpack --enable-libwebp --enable-libx265 --enable-libxml2 --enable-libxvid --enable-libzmq --enable-libzvbi --enable-lv2 --enable-omx --enable-openal --enable-opencl --enable-opengl --enable-sdl2 --enable-libdc1394 --enable-libdrm --enable-libiec61883 --enable-nvenc --enable-chromaprint --enable-frei0r --enable-libx264 --enable-shared
  libavutil      56. 31.100 / 56. 31.100
  libavcodec     58. 54.100 / 58. 54.100
  libavformat    58. 29.100 / 58. 29.100
  libavdevice    58.  8.100 / 58.  8.100
  libavfilter     7. 57.100 /  7. 57.100
  libavresample   4.  0.  0 /  4.  0.  0
  libswscale      5.  5.100 /  5.  5.100
  libswresample   3.  5.100 /  3.  5.100
  libpostproc    55.  5.100 / 55.  5.100
[libaom-av1 @ 0x555f5f662380] v1.0.0
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from 'Big_Buck_Bunny_720_10s_10MB_AV1.mp4':
  Metadata:
    major_brand     : isom
    minor_version   : 512
    compatible_brands: isomav01iso2mp41
    title           : Big Buck Bunny, Sunflower version
    artist          : Blender Foundation 2008, Janus Bager Kristensen 2013
    composer        : Sacha Goedegebure
    encoder         : Lavf58.63.100
    comment         : Creative Commons Attribution 3.0 - http://bbb3d.renderfarming.net
    genre           : Animation
  Duration: 00:00:10.00, start: 0.000000, bitrate: 8440 kb/s
    Stream #0:0(und): Video: av1 (Main) (av01 / 0x31307661), yuv420p(tv), 1280x720, 8437 kb/s, SAR 1:1 DAR 16:9, 30 fps, 30 tbr, 16k tbn, 16k tbc (default)
    Metadata:
      handler_name    : GPAC ISO Video Handler
[libaom-av1 @ 0x555f5f6674c0] v1.0.0
Stream mapping:
  Stream #0:0 -> #0:0 (av1 (libaom-av1) -> h264 (libx264))
Press [q] to stop, [?] for help
```
libaom-av1 is in my local version but not in the container version, so I imagine that is causing the issue.

**Comment by subdavis** [comment]
@BryonLewis thanks for doing this!  Do you think we could get that codec installed in the container, or is this worth wontfixing?

**Comment by BryonLewis** [comment]
I think it's worth fixing because 'AV1 is the codec of the future' or something for the last 5 years.  Either compile from source with the proper flags :nauseated_face:, or use a third party ppa install it (savoury1 seems to be popular).  I tried the third party `ppa:savoury1/ffmpeg4` using -it and afterwards it converted the file successfully. 

This brings up another issue in terms of #855, I believe the built in version of ffmpeg would have no chance of supporting AV1 so you couldn't use the source video or would have to require viame base to update it's version.

**Comment by subdavis** [comment]
Decision: install new PPA for support.  We are not going to pass original yet.
